File: merge_intervals/conflicting-appointments.js
class Interval {
    constructor(start, end) {
        this.start = start;
        this.end = end;
    }

    print_interval() {
        console.log("[" + this.start + ", " + this.end + "]");
    }
}

const can_attend_all_appointments = (intervals) => {
    intervals.sort((a, b) => a.start - b.start);

    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i].start < intervals[i - 1].end) return false;
    }

    return true;
}

console.log(`Can attend all appointments: ${can_attend_all_appointments([
    new Interval(1, 4),
    new Interval(2, 5),
    new Interval(7, 9),
])}`); // false

console.log(`Can attend all appointments: ${can_attend_all_appointments([
    new Interval(6, 7),
    new Interval(2, 4),
    new Interval(8, 12)
])}`); // true

console.log(`Can attend all appointments: ${can_attend_all_appointments([
    new Interval(4, 5),
    new Interval(2, 3),
    new Interval(3, 6)
])}`); // false

File: merge_intervals/python/conflicting-appointments.py
def can_attend_all_appointments(intervals):
    intervals.sort(key=lambda a: a[0])

    for i in range(1, len(intervals)):
        if intervals[i][0] < intervals[i - 1][1]:
            return False

    return True


if __name__ == '__main__':
    print(f"Can attend all appointments: {can_attend_all_appointments([[1, 4], [2, 5], [7, 9]])}")
    print(f"Can attend all appointments: {can_attend_all_appointments([[6, 7], [2, 4], [8, 12]])}")
    print(f"Can attend all appointments: {can_attend_all_appointments([[4, 5], [2, 3], [3, 6]])}")
